Automattic/jetpack
# My Jetpack: reduxify purchases data

This PR _reduxses_ the purchases data. Currently, the data is stored at the component level.

Fixes #

#### Changes proposed in this Pull Request:

*

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Easier once https://github.com/Automattic/jetpack/pull/22356 merges
* Test with a site with some purchases / plans
* Go to My Jetpack dashboard.
* Confirm you see the plans list in the Plans Section
* Check the redux implementation, filtering by `my-jetpack`

<img src="https://user-images.githubusercontent.com/77539/149786854-6a385185-2069-4683-b4b8-2dbbb75613c9.png" width="600px" />

<img src="https://user-images.githubusercontent.com/77539/149818718-c03db560-71df-4517-833c-bbe39b92a5bb.png" width="600px" />


* Check the client hits the `/purchases` endpoint
<img src="https://user-images.githubusercontent.com/77539/149787017-92be5319-2818-428d-9706-77871ba2ff49.png" width="600px" />

#### Tests

```cli
> jetpack test packages/my-jetpack js
```

<img src="https://user-images.githubusercontent.com/77539/149822984-7fcfd0a9-3b23-4b06-a798-44cd58f56e39.png" width="600px" />
